### ISSUE TITLE
Still getting Index out of range error calling GetPlayerMap

### ISSUE THREAD

**Issue by moddyz** [pogodevorg/pogo#16]
Still getting the Index out of range error in sessions.go, when accessing one of the Returns slice at index 0

This is the exact code:

```
package main

import (
    "log"

    "golang.org/x/net/context"

    "github.com/pkmngo-odi/pogo/api"
    "github.com/pkmngo-odi/pogo/auth"
)

func main() {

    ctx := context.Background()

    provider, err := auth.NewProvider("ptc", "xqaYVxmzdVD8AjS", "EGi9tUf8cmIOq9a")
    if err != nil {
        log.Println(err)
        return
    }

    location := &api.Location{
        Lon: 0.0,
        Lat: 0.0,
        Alt: 0.0,
    }

    feed := &api.VoidFeed{}

    session := api.NewSession(provider, location, feed, false)
    err = session.Init(ctx)
    if err != nil {
        log.Println(err)
        return
    }

    _, err = session.GetPlayer(ctx)
    if err != nil {
        log.Println(err)
        return
    }

    _, err = session.GetPlayerMap(ctx)
    if err != nil {
        log.Println(err)
        return
    }

    _, err = session.Announce(ctx)
    if err != nil {
        log.Println(err)
        return
    }
}
```

The stacktrace:

```
Moddy:~/go/src/github.com/moddyz/pokemapgo$ go build -v . && ./pokemapgo
github.com/moddyz/pokemapgo
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x417620, 0xc82000a0b0)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/pkmngo-odi/pogo/api.(*Session).GetPlayerMap(0xc820012640, 0x8951a0, 0xc82000a2a8, 0xc82042c2c0, 0x0, 0x0)
    /Users/Moddy/go/src/github.com/pkmngo-odi/pogo/api/session.go:230 +0x5e8
main.main()
    /Users/Moddy/go/src/github.com/moddyz/pokemapgo/main.go:43 +0x4c7
```

**Comment by zeevallin** [comment]
@moddyz Still having issues?

**Comment by femot** [comment]
I'm getting the error. I think it's caused by too many requests per second or at the same time.

Dirty fix from me:

```
    mapObjects = &protos.GetMapObjectsResponse{}
    if(len(response.Returns) > 5) {
        err = proto.Unmarshal(response.Returns[5], mapObjects)
        if err != nil {
            return nil, &ErrResponse{err}
        }
    } else {
        return nil, errors.New("Empty response")
    }
```

**Comment by femot** [comment]
Not getting the error anymore after staggering the requests to 1 per second

**Comment by zeevallin** [comment]
@femot So if you're doing more than 1 requests per second, you start getting this error again? Reeks of concurrency issues. Mind looking into it and try a few different things?

**Comment by femot** [comment]
@zeeraw Didn't try to find out the breakpoint, but 1s seems to work fine for now.

@moddyz code is a good example because he calls `session.GetPlayer(ctx)` in rapid succession without concurrency.

I also had the problem sometimes after startup while using 1 account (rarely). After switching tests to 5 accounts it happened all the time (they were calling `session.GetPlayer(ctx)` simultaneously before i fixed it).

If you want I can try a few things, but I thought I read something about this being an issue on Niantics side where they just say "nope" if you're doing too much - not 100% sure though.

The code I used can be found here: https://github.com/femot/spawnfinder

**Comment by femot** [comment]
@zeeraw I did some testing with this: https://gist.github.com/femot/685f4796c090de5a6ee85e08d4c0dcf3
Seems like the magic value is around 120ms.

Edit: Do not recommend testing this if you want to scan something. This can get your IP to be softbanned :cry:

**Comment by femot** [comment]
fixed in ec8a3f26e4dde20585f6e0329501413bbc1673a4
